### PR TITLE
refactor AsyncDependenciesBlock to es6 syntax

### PR DESCRIPTION
**Pull request by ritikrishu**
**What kind of change does this PR introduce?**
* refactoring, ES5 => ES6
* replace trailing ',' from chunks ids string passed to `hash()` function in `updateHash()`, if any.

**Did you add tests for your changes?**
All tests are passing.

**If relevant, link to documentation update:**
N/A

**Summary**
Upgrade the AsyncDependenciesBlock to ES6

**Does this PR introduce a breaking change?**
NO

### CODE REVIEW

File: lib/AsyncDependenciesBlock.js [hunk 1]
```diff
@@ -2,61 +2,55 @@
...
+module.exports = class AsyncDependenciesBlock extends DependenciesBlock {
+	constructor(name, module, loc) {
+		super();
+		this.chunkName = name;
+		this.chunks = null;
+		this.module = module;
+		this.loc = loc;
+
+		Object.defineProperty(this, "chunk", {
+			get: function() {
```
Comment (abouthiroppy) [inline lib/AsyncDependenciesBlock.js]: You can use Arrow Function 😉

File: lib/AsyncDependenciesBlock.js [hunk 2]
```diff
@@ -2,61 +2,55 @@
...
+				throw new Error("`chunk` was been renamed to `chunks` and is now an array");
 			}
 		});
 	}
-};
+	updateHash(hash) {
+		hash.update(this.chunkName || "");
+		hash.update(this.chunks && this.chunks.map((chunk) => {
+			return typeof chunk.id === "number" ? chunk.id : "";
+		}).join(",").replace(/,\s*$/, "") || "");
```
Comment (sokra) [inline lib/AsyncDependenciesBlock.js]: Why adding this replace call here?

File: lib/AsyncDependenciesBlock.js [hunk 3]
```diff
@@ -2,61 +2,55 @@
...
+
+module.exports = class AsyncDependenciesBlock extends DependenciesBlock {
+	constructor(name, module, loc) {
+		super();
+		this.chunkName = name;
+		this.chunks = null;
+		this.module = module;
+		this.loc = loc;
+
+		Object.defineProperty(this, "chunk", {
```
Comment (sokra) [inline lib/AsyncDependenciesBlock.js]: You can use

``` js
get chunk() { ... }
set chunk() { ... }
```

in the class

File: lib/AsyncDependenciesBlock.js (hunk 1 above)
Comment (ritikrishu) [inline lib/AsyncDependenciesBlock.js]: @abouthiroppy done https://github.com/webpack/webpack/pull/3850/commits/5b236b9b41ed5db78499042b48f1065684bace7a

File: lib/AsyncDependenciesBlock.js (hunk 3 above)
Comment (ritikrishu) [inline lib/AsyncDependenciesBlock.js]: @sokra done https://github.com/webpack/webpack/pull/3850/commits/5b236b9b41ed5db78499042b48f1065684bace7a

File: lib/AsyncDependenciesBlock.js (hunk 2 above)
Comment (ritikrishu) [inline lib/AsyncDependenciesBlock.js]: `return typeof chunk.id === "number" ? chunk.id : "";`
look at this line I figured chunk.id can be a non-number entity, so if
`this.chunks = [{id: 1, ...}, {id: 2,...}, {id: 'abc',...}]`

so in this case 
```
this.chunks && this.chunks.map((chunk) => {
+			return typeof chunk.id === "number" ? chunk.id : "";
+		}).join(",") || ""
```
should return `1,2,`
while
```
this.chunks && this.chunks.map((chunk) => {
+			return typeof chunk.id === "number" ? chunk.id : "";
+		}).join(",").replace(/,\s*$/, "") || ""
```
will return `1,2` <-removing the trailing ",".
@sokra If this change is not relevant in context, I'll revert the `replace` addition.

**Pull request merged**